gnomad_qc/v3/create_release/prepare_internal_data_release.py

```diff
@@ -0,0 +1,398 @@
+    row_fields = set([field for field in global_freq_ht.row_value]).difference(
+        set([field for field in freq_ht.row_value])
+    )
+    logger.info(
+        f"Adding back the following row annotations onto concatenated frequencies: {row_fields}"
+    )
+    freq_ht = freq_ht.annotate(**global_freq_ht[freq_ht.key].select(*row_fields))
+
+    global_fields = set([field for field in global_freq_ht.globals]).difference(
+        set([field for field in freq_ht.globals])
+    )
+    global_fields.remove("downsamplings")
+    logger.info(
+        f"Adding back the following global annotations onto concatenated frequencies: {global_fields}"
+    )
+    freq_ht = freq_ht.annotate_globals(
+        **global_freq_ht.index_globals().select(*global_fields)
+    )
+
+    logger.info("Preparing release Table annotations...")
+    ht = prepare_annotations(freq_ht, args.model_id, test=False)
+
+    logger.info("Removing chrM and sites without filter...")
+    ht = hl.filter_intervals(ht, [hl.parse_locus_interval("chrM")], keep=False)
+    ht = ht.filter(hl.is_defined(ht.filters), keep=True)
+
+    # Add in new annotations for clinical variant interpretation
+    in_silico_ht = hl.read_table(
+        "gs://gnomad/annotations/hail-0.2/ht/genomes_v3.1/gnomad_genomes_v3.1.analyst_annotations.ht"
+    )  # TODO: update path with versioned Table resource path when available
+    ht = ht.annotate(**in_silico_ht[ht.key])
+
+    # Splice in fix to set female metrics to NA on Y chr
+    female_idx = [
+        hl.eval(ht.freq_index_dict.get(x))
+        for x in hl.eval(ht.freq_index_dict)
+        if "XX" in x
+    ]
+    freq_idx_range = hl.eval(hl.range(hl.eval(hl.len(hl.eval(ht.freq_meta)))))
+    null_y_struct = hl.struct(
+        AC=hl.null(hl.tint32),
+        AF=hl.null(hl.tfloat64),
+        AN=hl.null(hl.tint32),
+        homozygote_count=hl.null(hl.tint32),
+    )
+    ht = ht.annotate(
+        freq=hl.if_else(
+            (ht.locus.in_y_nonpar() | ht.locus.in_y_par()),
+            hl.map(
+                lambda x: hl.if_else(
+                    hl.array(female_idx).contains(x), null_y_struct, ht.freq[x]
+                ),
+                freq_idx_range,
+            ),
+            ht.freq,
+        )
+    )
+
+    ht = ht.checkpoint(
+        "gs://gnomad-tmp/release/v3.1/gnomad.genomes.v3.1.sites.ht"
+        if args.test
+        else release_ht_path(public=False),
+        args.overwrite,
+    )
+    logger.info(f"Final variant count: {ht.count()}")
+    ht.describe()
+    ht.show()
+    ht.summarize()
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--test", help="Runs a test on chr20:1-1000000", action="store_true"
+    )
+    parser.add_argument(
+        "--model_id",
+        help="Filtering model ID to use",
+        default="vqsr_alleleSpecificTrans",
+    )
+    parser.add_argument(
+        "--n_partitions",
```

Comment:
parameter not used

Reply:
Is this something that would generally be useful to have? I'm happy to add a final repartition if you think that's a desirable feature to include.

Reply:
I'm honestly not the best person to ask on this, @ch-kr you have recently thought more about the partitioning, do you have any thoughts?